src/test/java/ai/datahunters/md/server/server/photos/UploadPhotosEndpointTest.java

```diff
@@ -56,4 +63,43 @@ public void uploadPhotos() throws IOException {
                 );
 
     }
+
+
+    @Test
+    public void probeArchiveAndExtract() throws IOException, CompressorException {
+        List<String> expected = Arrays.asList("smile.png");
+
+        Assertions.assertThrows(IOException.class,
+                () -> archiveHandler.probeContentAndUnarchive(openArchive("uploadendpointtest/test_file.zip")));
+        Assertions.assertThrows(IOException.class,
+                () -> archiveHandler.probeContentAndUnarchive(openArchive("uploadendpointtest/CORRUPTED_ZIP.zip")));
```

Comment:
You could alo check if the directory is really removed if something goes wrong with unpacking

Reply:
I cannot think of any stritght forward way to do this using mockito, so I'd have to handcraft a tar/zip with precisely moddified  entry header. I'll try.